source/core_mqtt.c

```diff
@@ -995,29 +995,38 @@ static MQTTStatus_t handleKeepAlive( MQTTContext_t * pContext )
 {
     MQTTStatus_t status = MQTTSuccess;
     uint32_t now = 0U, keepAliveMs = 0U;
+    uint32_t pingrespTimeoutMs = MQTT_PINGRESP_TIMEOUT_MS;
 
     assert( pContext != NULL );
     assert( pContext->getTime != NULL );
 
     now = pContext->getTime();
     keepAliveMs = 1000U * ( uint32_t ) pContext->keepAliveIntervalSec;
 
+    /* It is not useful for the ping response timeout to be greater than half
+     * the keep alive, as the server will have disconnected us by that point. */
+    if( pingrespTimeoutMs > ( keepAliveMs / 2 ) )
```

Comment:
Should this be prevented as a compile time build failure instead of changing it at runtime and the customer not being aware of the use of a different response timeout than their configured value?

Reply:
I don't see how this could be done at compile time, since the keep alive information is set in the connect info that's passed to `MQTT_Connect()`. The best that could be done is a warning log